src/Components/ADT3DViewer/ADT3DViewer.tsx

```diff
@@ -481,12 +505,23 @@ const ADT3DViewerBase: React.FC<IADT3DViewerProps> = ({
                     coloredMeshes.push(sceneColoredMeshItem);
                 });
             });
-            setAlertBadges(sceneAlerts);
             setColoredMeshItems(coloredMeshes);
         }
-    }, [sceneVisuals, coloredMeshItemsProp, sceneAlerts]);
+    }, [sceneVisuals, coloredMeshItemsProp]);
+
+    useEffect(() => {
+        if (sceneAlerts) {
+            const markers: Marker[] = [];
+            sceneAlerts.forEach((alert) => {
+                const badge = createBadge(alert);
+                markers.push(badge);
+            });
+
+            setMarkers(markers);
+        }
+    }, [sceneAlerts, sceneThemeState.sceneBackground]);
```

Comment:
let's not fake out the dependency list, if something insde depends on the sceneBakcground, please pass it through to the function instead of adding it to the dependency list here without actually using it in this effect directly. It just runs too much of a risk of being cleaned up at some point since we can't see that it's used deeper in the pipeline.

Reply:
I have added a comment, we have this as a dependency for the useEffect as we need it to rerun when the background changes to update the badge colors.